File: Source/ui_android/java/com/virtualapplications/play/VirtualPadItem.java
package com.virtualapplications.play;

import android.graphics.*;

class VirtualPadItem
{
	private int _pointerId = -1;
	protected RectF _bounds;
	
	public VirtualPadItem(RectF bounds)
	{
		_bounds = bounds;
	}
	
	public void setPointerId(int pointerId)
	{
		_pointerId = pointerId;
	}
	
	public int getPointerId()
	{
		return _pointerId;
	}
	
	public RectF getBounds()
	{
		return _bounds;
	}
	
	public void onPointerDown(float x, float y)
	{
		
	}
	
	public void onPointerUp()
	{
		
	}
	
	public void onPointerMove(float x, float y)
	{
		
	}
	
	public void draw(Canvas canvas)
	{
		
	}
}

File: Source/ui_win32/VirtualPad/VirtualPadItem.cpp
#include "VirtualPadItem.h"

CVirtualPadItem::~CVirtualPadItem()
{

}

void CVirtualPadItem::OnMouseDown(int, int)
{

}

void CVirtualPadItem::OnMouseMove(int, int)
{

}

void CVirtualPadItem::OnMouseUp()
{

}

void CVirtualPadItem::SetPadHandler(CPH_Generic* padHandler)
{
	m_padHandler = padHandler;
}

Gdiplus::RectF CVirtualPadItem::GetBounds() const
{
	return m_bounds;
}

void CVirtualPadItem::SetBounds(const Gdiplus::RectF& bounds)
{
	m_bounds = bounds;
}

void CVirtualPadItem::SetImage(Gdiplus::Bitmap* image)
{
	m_image = image;
}

uint32 CVirtualPadItem::GetPointerId() const
{
	return m_pointerId;
}

void CVirtualPadItem::SetPointerId(uint32 pointerId)
{
	m_pointerId = pointerId;
}
